frappe/event_streaming/doctype/event_producer/test_event_producer.py

```diff
@@ -198,6 +187,84 @@ def get_remote_site(self):
 		)
 		return producer_site
 
+	def test_mapping(self):
+		event_producer = frappe.get_doc('Event Producer', self.producer_url)
+		event_producer.producer_doctypes = []
+		mapping = [{
+			'local_fieldname': 'description',
+			'remote_fieldname': 'content'
+		}]
+		event_producer.append('producer_doctypes', {
+			'ref_doctype': 'ToDo',
+			'use_same_name': 1,
+			'has_mapping': 1,
+			'mapping': get_mapping('ToDo to Note', 'ToDo', 'Note', mapping)
+		})
+		event_producer.save()
+		producer = self.get_remote_site()
+		producer_note = frappe.get_doc(dict(doctype='Note', title='Test Mapping', content='Test Mapping'))
+		delete_on_remote_if_exists(producer, 'Note', {'title': producer_note.title})
+		producer_note = producer.insert(producer_note)
+		self.pull_producer_data()
+		#check inserted
+		self.assertTrue(frappe.db.exists('ToDo', {'description': producer_note.content}))
+
+		#update in producer
+		producer_note['content'] = 'test mapped doc update sync'
+		producer_note = producer.update(producer_note)
+		self.pull_producer_data()
+		time.sleep(5)
```

Comment:
Sometimes there is a delay in docs creation if data is huge. If database checks are made immediately they fail because the remote doc is not synced yet.